gpytorch/kernels/periodic_kernel.py

```diff
@@ -107,10 +113,14 @@ def _set_period_length(self, value):
         self.initialize(raw_period_length=self.raw_period_length_constraint.inverse_transform(value))
 
     def forward(self, x1, x2, diag=False, **params):
-        x1_ = x1.div(self.period_length)
-        x2_ = x2.div(self.period_length)
-        diff = self.covar_dist(x1_, x2_, diag=diag, **params)
-        res = torch.sin(diff.mul(math.pi)).pow(2).mul(-2 / self.lengthscale).exp_()
+        if x1.dim() == 3 and x2.dim() == 3:  # batched operation
```

Comment:
This code will fail if you have a multi-batch GP (4 or more tensor dimensions). It's better to do `diff = x1.unsqueeze(-2) - x2.unsqueeze(-3)`.